src/Components/Components/src/Rendering/Renderer.cs

```diff
@@ -636,8 +640,8 @@ private async Task RemoveEventHandlerIdsAsync(ArrayRange<int> eventHandlerIds, T
                     // That remains the caller's business.
                 }
 
-                // We know the next execution will complete synchronously, so nothing to await
-                _ = RemoveEventHandlerIdsAsync(eventHandlerIdsClone, Task.CompletedTask);
+                // We know the next execution will complete synchronously, so no infinite loop
+                RemoveEventHandlerIds(eventHandlerIdsClone, Task.CompletedTask);
```

Comment:
If I follow this correctly, all this would do is call

```C#
var array = eventHandlerIds.Array;
                var count = eventHandlerIds.Count;
                for (var i = 0; i < count; i++)
                {
                    _eventBindings.Remove(array[i]);
                }
```

Should we just inline that here or another local function rather than recursively call this? Makes it much easier to follow

Reply:
It's factored out into a local so that the outer method isn't `async` and hence doesn't come with all the state machine overhead (which includes an allocation to hold hoisted locals).

Reply:
Oh wait I see what you mean. TBH I think it's fine either way. In the interests of getting this merged I plan to leave as-is.